### PR TITLE
tiltfile: allow live_update syncs from dependent images

### PR DESCRIPTION
**Pull request by nicks**
Hello @maiamcc,

Please review the following commits I made in branch nicks/ch2351/liveupdate:

3093f8a14e4e2eb3074702a33a86962a64c6b386 (2019-04-29 15:36:42 -0400)
tiltfile: allow live_update syncs from dependent images
also add integration tests

### CODE REVIEW

File: integration/live_update_base_image/Tiltfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+k8s_resource_assembly_version(2)
```
Comment (maiamcc) [inline integration/live_update_base_image/Tiltfile]: pretty sure this is no longer necessary

File: internal/tiltfile/live_update_test.go [hunk 2]
```diff
@@ -184,6 +184,42 @@ docker_build('gcr.io/foo', 'foo',
...
+
+	f.gitInit("")
+	f.file("a/message.txt", "message")
+	f.file("imageA.dockerfile", `FROM golang:1.10
+ADD message.txt /src/message.txt
+`)
+	f.file("imageB.dockerfile", "FROM gcr.io/image-a")
+	f.yaml("foo.yaml", deployment("foo", image("gcr.io/image-b")))
+	f.file("Tiltfile", `
+k8s_resource_assembly_version(2)
```
Comment (maiamcc) [inline internal/tiltfile/live_update_test.go]: no longer necessary?

File: internal/tiltfile/tiltfile_state.go [hunk 3]
```diff
@@ -757,9 +767,58 @@ func (s *tiltfileState) checkForImpossibleLiveUpdates(m model.Manifest) {
...
+
+		err = s.validatedLiveUpdate(iTarg, g)
+		if err != nil {
+			return err
 		}
 	}
+	return nil
+}
+
+func (s *tiltfileState) validatedLiveUpdate(iTarget model.ImageTarget, g model.TargetGraph) error {
```
Comment (maiamcc) [inline internal/tiltfile/tiltfile_state.go]: now that this isn't a getter (😅), should be `validateLiveUpdate`

File: integration/live_update_base_image/Tiltfile (hunk 1 above)
Comment (nicks) [reply integration/live_update_base_image/Tiltfile]: done

File: internal/tiltfile/live_update_test.go (hunk 2 above)
Comment (nicks) [reply internal/tiltfile/live_update_test.go]: done

File: internal/tiltfile/tiltfile_state.go (hunk 3 above)
Comment (nicks) [reply internal/tiltfile/tiltfile_state.go]: done

**Comment by nicks** [comment]
currently this breaks the integration tests, due to the bug fixed in https://github.com/windmilleng/tilt/pull/1554

**Pull request merged**